### ISSUE TITLE
APNS:216 CMD: 8 Status: Invalid token ID: 0

### ISSUE THREAD

**Issue by tfohlmeister** [airnotifier/airnotifier#126]
I receive the above error on a production server when sending a broadcast through both the API as well as the web interface.
It seems like I have a misconfigured token in my long list of devices. Why does it stop broadcasting completely? How to find that token and delete it?

**Comment by pathikb** [comment]
I ran into the same issue. Seems like either you have an invalid token or a Sandbox token being used on Production. Sadly the solution I used will need a little bit or trial and error. This is what I used to fix this:

1. Extracted all tokens from MongoDB database and converted it to a plist
2. Wrote a dummy app which reads from plist and pushes notifications (push not broadcast) to them one by one.
3. Watch which ones fail, remove them from Plist (make a note of this token elsewhere) and re-run the process.
4. Remove all invalid tokens from MongoDB
